File: src/test/java/no/nav/arbeidsgiver/sykefravarsstatistikk/api/AssertUtils.java
package no.nav.arbeidsgiver.sykefravarsstatistikk.api;

import java.math.BigDecimal;
import java.math.RoundingMode;

import static org.assertj.core.api.Assertions.assertThat;

public class AssertUtils {

    public static void assertBigDecimalIsEqual(BigDecimal actual, float expected) {
        assertThat(actual.setScale(6, RoundingMode.HALF_UP))
                .isEqualTo(BigDecimal.valueOf(expected).setScale(6, RoundingMode.HALF_UP));
    }


}

File: src/test/kotlin/no/nav/arbeidsgiver/sykefravarsstatistikk/api/AssertUtils.kt
package no.nav.arbeidsgiver.sykefravarsstatistikk.api

import org.assertj.core.api.Assertions
import java.math.BigDecimal
import java.math.RoundingMode

object AssertUtils {
    const val SCALE = 6

    fun assertBigDecimalIsEqual(actual: BigDecimal?, expected: BigDecimal?) {
        if (actual == null || expected == null) {
            Assertions.assertThat(actual).isEqualTo(expected)
            return
        }
        Assertions.assertThat(actual.setScale(SCALE, RoundingMode.HALF_UP))
            .isEqualTo(expected.setScale(SCALE, RoundingMode.HALF_UP))
    }
}
